File: core/src/net/brenig/pixelescape/render/ui/general/PixelDialog.java
package net.brenig.pixelescape.render.ui.general;

import com.badlogic.gdx.scenes.scene2d.ui.Cell;
import com.badlogic.gdx.scenes.scene2d.ui.Dialog;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.badlogic.gdx.utils.Align;

/**
 * Dialog for PixelEscape<br></br>
 * Provides a texture and some default settings
 */
public class PixelDialog extends Dialog {

	private static final float defaultFontScale = 0.7F;

	private float prefWidth = -1;
	private float prefHeight = -1;


	public PixelDialog(String title, Skin skin) {
		super(title, skin);
	}

	public PixelDialog(String title, Skin skin, String windowStyleName) {
		super(title, skin, windowStyleName);
	}

	public PixelDialog(String title, WindowStyle windowStyle) {
		super(title, windowStyle);
	}

	public void setPrefHeight(float prefHeight) {
		this.prefHeight = prefHeight;
	}

	public void setPrefWidth(float prefWidth) {
		this.prefWidth = prefWidth;
	}

	public void init() {
		getTitleLabel().setAlignment(Align.center);
		padTop(50);
		padBottom(10);
		getContentTable().left().padTop(10);
		getButtonTable().padTop(10);
		getColor().a = 0;
		invalidateHierarchy();
		invalidate();
		layout();
	}

	private void updateCells(float width) {
		if(getContentTable() != null) {
			for (Cell c : getContentTable().getCells()) {
				c.width(width);
			}
		}
	}

	public Label label(String text) {
		Label l = new Label(text, getSkin());
		l.setWrap(true);
		l.setFontScale(defaultFontScale);
		getContentTable().add(l).width(getPrefContentWidth()).row();
		return l;
	}

	public float getPrefContentWidth() {
		if(prefWidth != -1) return prefWidth;
		return getTitleLabel().getPrefWidth();
	}

	@Override
	public float getPrefHeight() {
		if(prefHeight != -1) return prefHeight;
		return super.getPrefHeight();
	}

	@Override
	public float getPrefWidth() {
		if(prefWidth != -1) return prefWidth + getPadLeft() + getPadRight();
		return super.getPrefWidth();
	}

	public TextButton buttonYes(ClickListener listener) {
		TextButton btnYes = new TextButton("Yes", getSkin());
		btnYes.addListener(listener);
		button(btnYes);
		return btnYes;
	}

	public TextButton buttonNo(ClickListener listener) {
		TextButton btnNo = new TextButton("No", getSkin());
		btnNo.addListener(listener);
		button(btnNo);
		return btnNo;
	}
}

File: core/src/net/brenig/pixelescape/render/ui/general/PixelDialog.kt
package net.brenig.pixelescape.render.ui.general

import com.badlogic.gdx.scenes.scene2d.ui.*
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.badlogic.gdx.utils.Align

/**
 * Dialog for PixelEscape<br></br>
 * Provides a texture and some default settings
 */
class PixelDialog : Dialog {

    private var prefWidth = -1f
    private var prefHeight = -1f

    val prefContentWidth: Float
        get() = if (prefWidth != -1f) prefWidth else titleLabel.prefWidth


    constructor(title: String, skin: Skin) : super(title, skin)

    constructor(title: String, skin: Skin, windowStyleName: String) : super(title, skin, windowStyleName)

    constructor(title: String, windowStyle: Window.WindowStyle) : super(title, windowStyle)

    fun setPrefHeight(prefHeight: Float) {
        this.prefHeight = prefHeight
    }

    fun setPrefWidth(prefWidth: Float) {
        this.prefWidth = prefWidth
    }

    fun init() {
        titleLabel.setAlignment(Align.center)
        padTop(50f)
        padBottom(10f)
        contentTable.left().padTop(10f)
        buttonTable.padTop(10f)
        color.a = 0f
        invalidateHierarchy()
        invalidate()
        layout()
    }

    private fun updateCells(width: Float) {
        if (contentTable != null) {
            for (c in contentTable.cells) {
                c.width(width)
            }
        }
    }

    fun label(text: String): Label {
        val l = Label(text, skin)
        l.setWrap(true)
        l.setFontScale(defaultFontScale)
        contentTable.add(l).width(prefContentWidth).row()
        return l
    }

    override fun getPrefHeight(): Float {
        return if (prefHeight != -1f) prefHeight else super.getPrefHeight()
    }

    override fun getPrefWidth(): Float {
        return if (prefWidth != -1f) prefWidth + padLeft + padRight else super.getPrefWidth()
    }

    fun buttonYes(listener: ClickListener): TextButton {
        val btnYes = TextButton("Yes", skin)
        btnYes.addListener(listener)
        button(btnYes)
        return btnYes
    }

    fun buttonNo(listener: ClickListener): TextButton {
        val btnNo = TextButton("No", skin)
        btnNo.addListener(listener)
        button(btnNo)
        return btnNo
    }

    companion object {

        private val defaultFontScale = 0.7f
    }
}
